libdebug/architectures/amd64/amd64_stack_unwinder.py

```diff
@@ -51,3 +52,61 @@ def unwind(self, target: "ThreadContext") -> list:
                 break
 
         return stack_trace
+
+    def get_return_address(self, target: "ThreadContext") -> int:
+        """
+        Get the return address of the current function.
+
+        Args:
+            target (ThreadContext): The target ThreadContext.
+
+        Returns:
+            int: The return address.
+        """
+
+        # Set the thread to background mode to avoid deadlock
+        target._in_background_op = True
+
+        instruction_window = target.memory[target.rip, 4]
+
+        # Check if the instruction window is a function preamble and handle each case
+        return_address = None
+
+        if self._preamble_state(instruction_window) == 0:
+            return_address = target.memory[target.rbp + 8, 4]
```

Comment:
I did not get why it reads only 4 bytes here instead of 8. In your tests, the addresses were all similar to 0x401XXX, hence they are represented with less than 4 bytes. However, clearly, this is not always the case.

Reply:
You are correct, this was an oversight.